dask/array/tests/test_array_core.py

```diff
@@ -870,6 +871,75 @@ def test_store():
     assert raises(ValueError, lambda: store([at, bt], [at, bt]))
 
 
+def test_store_compute_false():
+    d = da.ones((4, 4), chunks=(2, 2))
+    a, b = d + 1, d + 2
+
+    at = np.zeros(shape=(4, 4))
+    bt = np.zeros(shape=(4, 4))
+
+    v = store([a, b], [at, bt], compute=False)
+    assert (at == 0).all() and (bt == 0).all()
+    v.compute()
+    assert (at == 2).all() and (bt == 3).all()
+
+
+class ThreadSafetyError(Exception):
+    pass
+
+
+class NonthreadSafeStore(object):
+    def __init__(self):
+        self.in_use = False
+
+    def __setitem__(self, key, value):
+        if self.in_use:
+            raise ThreadSafetyError()
+        self.in_use = True
+        time.sleep(0.001)
+        self.in_use = False
+
+
+class ThreadSafeStore(object):
+    def __init__(self):
+        self.concurrent_uses = 0
+        self.max_concurrent_uses = 0
+
+    def __setitem__(self, key, value):
+        self.concurrent_uses += 1
+        self.max_concurrent_uses = max(self.concurrent_uses, self.max_concurrent_uses)
+        time.sleep(0.01)
+        self.concurrent_uses -= 1
+
+
+def test_store_locks():
```

Comment:
Are you sure this will always be run with multi-threaded dask, even on a single core machine? I would set the number of threads manually for this test, just to be safe.